pkg/task/get.go

```diff
@@ -1,31 +1,57 @@
 package task
 
 import (
-	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
 
-	"github.com/puppetlabs/nebula-sdk/pkg/taskutil"
 	"github.com/puppetlabs/nebula-sdk/pkg/workflow/spec/evaluate"
 )
 
 func (ti *TaskInterface) ReadData(path string) ([]byte, error) {
-	eval, err := taskutil.EvaluatorFromDefaultPlan(ti.opts)
+	u, err := url.Parse(ti.opts.SpecURL)
 	if err != nil {
 		return nil, err
 	}
 
+	q, err := url.ParseQuery(u.RawQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	if path != "" {
+		q.Add("q", path)
+		q.Add("lang", "jsonpath-template")
+
+		u.RawQuery = q.Encode()
+	}
+
+	resp, err := http.Get(u.String())
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode/100 != 2 {
+		return nil, nil
```

Comment:
Previously errors were returned if there was an error with `EvaluatorFromDefaultPlan` but `return nil, nil` happened if there was any error with `EvaluateQuery` or `EvaluateAll`. So previously in the case of failed spec loading an error would be returned, but in the case of failed evaluation, an empty spec would be returned.

The api would return a 500 task spec lookup, decoding, or evaluation error.

To match previous behavior, I should make `ni get` return an error in the cases of lookup or decoding, but not in an evaluation error case.

Reply:
Updated to use 500/404/422 return codes to handle this.